### PR TITLE
53 feature request add a calendar view of events

### PR DESCRIPTION
**Pull request by davidslusser**
## Pull Request

**Description:**
Closes issue 53. This change adds a monthly calendar view of events with events details available via modal. Also added a link to discord. 

spokanetechcalendar

**Related Issues:**
- #1
- #2

**Checklist:**
- [x] All tests pass.
- [x] Code follows the project's coding standards.
- [ ] Documentation has been updated.

### CODE REVIEW

File: src/web/views.py [hunk 1]
```diff
@@ -2,11 +2,15 @@
 from django.http import HttpRequest, HttpResponse
 from django.shortcuts import render
 from django.utils import timezone
+from django.template import loader
 
+# from django.urls import reverse
```
Comment (joeriddles) [inline src/web/views.py:7]: nit: remove comment out import

File: src/web/urls.py [hunk 2]
```diff
@@ -11,4 +11,6 @@
     path('groups/<int:pk>', views.DetailTechGroup.as_view(), name="get_tech_group"),
     path('events/<int:pk>', views.DetailEvent.as_view(), name="get_event"),   
     path('build_sidebar', views.BuildSidebar.as_view(), name="build_sidebar"),
+    path("get_event_details/<int:pk>", views.GetEventDetailsModal.as_view(), name="get_event_details"),
```
Comment (joeriddles) [inline src/web/urls.py:14]: I think this URL should be nested under `events/...`.

File: src/web/templates/web/partials/modal/detail_event.htm [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<div class="container-fluid my-3">
+    {% if object.group %}
```
Comment (joeriddles) [inline src/web/templates/web/partials/modal/detail_event.htm:2]: I'm not sure why we made the `group` FK nullable, seems like it should always be required... but that's outside the scope of the PR.

**Review by joeriddles — APPROVED** [review]
Love the UI! Nice tests. Can we change the subtitle of the calendar page to be "February, 2024" instead of "2/2024" (more human readable)?

File: src/web/templates/web/partials/modal/detail_event.htm (hunk 3 above)
Comment (davidslusser) [reply src/web/templates/web/partials/modal/detail_event.htm:2]: Yeah, I'd think that should be required. The template should reflect the model, so those so be resolved together.

**Comment by davidslusser** [comment]
> Love the UI! Nice tests. Can we change the subtitle of the calendar page to be "February, 2024" instead of "2/2024" (more human readable)?
> 
> image

using the text of the month instead of the digit is a good idea, but this comes from an included package; I'll make the update there.

File: src/web/urls.py (hunk 2 above)
Comment (davidslusser) [reply src/web/urls.py:14]: are you thinking like `events/<int:pk>/details` ?

Comment (joeriddles) [reply src/web/urls.py:14]: Yeah!

**Pull request merged**